### DISCUSSION TITLE
$GITHUB_OUTPUT: ambiguous redirect

### DISCUSSION THREAD

**Discussion by AswiniPriyanka** [community/community#36746] [Other Feature Feedback, Questions, & Ideas] [answered]
Getting the following error even when running the sample code 

jobs:
  job1:
    runs-on: [deere-ubuntu-latest]
    # Map a step output to a job output
    outputs:
      output1: ${{ steps.step1.outputs.test }}
      output2: ${{ steps.step2.outputs.test }}
    steps:
      - id: step1
        run: echo "test=hello" >> $GITHUB_OUTPUT
      - id: step2
        run: echo "test=world" >> $GITHUB_OUTPUT

**Comment by AswiniPriyanka** [comment, marked as answer]
Found the answer, need to use $env:GITHUB_OUTPUT instead of $GITHUB_OUTPUT

**Reply by cmitz** [reply]
Could you maybe explain why you tried this? What is this syntax and why is it different from the docs?

**Reply by Leee-xx** [reply]
I believe that's the Windows powershell syntax, though I'm not sure why that would resolve this issue when the workflow indicates an ubuntu runner.

**Reply by harrycain72** [reply]
Unfortunately using  >> $GITHUB_OUTPUT nor the proposed workaround $env:GITHUB_OUTPUT is working for me.

Using the first way throws the $GITHUB_OUTPUT: ambiguous redirect error.
Using the second way, the called wf works, but the caller does not get the value of the output.
Only using the deprecated way ::set-output works fine 

It is really frustrating ...


Here is my setup:

`name: Test_reusable_workflow

on:
  workflow_dispatch:
  workflow_call:
    # Map the workflow outputs to job outputs
    outputs:
      output:
        description: "The first output string"
        value: ${{ jobs.job-1.outputs.myoutput }}


jobs:
  job-1:
    name: name of job-1
    runs-on: [atc-ubuntu-20.04]
    outputs:
      myoutput: ${{ steps.step1.outputs.myoutput }}
    steps:
      - id: step1
        run: |
          echo "::set-output name=myoutput::myvalue"
  
  job-2:
    name: name of job-2
    runs-on: [atc-ubuntu-20.04]
    needs: job-1
    steps:
      - name: show output variable
        run: |
          echo "myoutput value: ${{ needs.job-1.outputs.myoutput }}" `

`name: Test_caller_workflow

on:
  workflow_dispatch:

jobs:
  job-1:
    name: name of job-1
    uses: ./.github/workflows/Test_WF_reusable.yml

  
  job-2:
    name: name of job-2
    runs-on: [ atc-ubuntu-20.04 ]
    needs: [ job-1 ]
    environment: sandbox

    steps:
      - run: |
         echo "Value of job output is: ${{ needs.job-1.outputs.output }} "
  `

**Reply by Leee-xx** [reply]
I got the `ambiguous redirect` error and it turned out that my runner was using an older runner version, which I believe has to be at least 2.297.0 to use the new syntax.

**Reply by harrycain72** [reply]
Hi @Leee-xx , in our company (using GHE we are using the following version for our runners .
Current runner version: '2.289.3'
With Github Enterprise 3.5.3 (now)
An upgrade to Github Enterprise 3.6.3 planned for tomorrow.
I will check the runner version after upgrade.
Thanx for the tip.

Docu related to this topic:
https://github.com/actions/runner/releases

[v2.297.0](https://github.com/actions/runner/releases/tag/v2.297.0)
Features
Created prerelease runner package for win-arm64 architecture (https://github.com/actions/runner/pull/2022)
Added GITHUB_STATE and GITHUB_OUTPUT environment file commands (https://github.com/actions/runner/pull/2118)

**Reply by Leee-xx** [reply]
@harrycain72 https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

> Starting today runner version 2.298.2 will begin to warn you if you use the save-state or set-output commands via stdout. 

> If you are using self-hosted runners make sure they are updated to version 2.297.0 or greater.

**Comment by ossanna16** [comment]
Thank you for asking a good question @AswiniPriyanka, I'm glad you found our community 🙂

**Comment by gutwrinch** [comment]
I wish I could speak your language but I'm just a construction worker who went on a platform one day and put in an amount of bitcoins to all the seats February the 5th 2022 since that day I've searched to the end of the Earth to figure out what happened to my money and I still don't have a clue I've narrowed it down to the blockchain and the address but finding it from there I don't know I have no idea how to code or speak the language of computers but I can give you an address and see if you can trace it back 1A1zp1eP5QGefi2dmPTfTL5SLmv7DivfNa--Sent from my Android phone with mail.com Mail. Please excuse my brevity.On 2022 10 24, Mon, 5:16 AM Casper Smits ***@***.***> wrote:

   
  Could you maybe explain why you tried this? What is this syntax and why is it different from the docs?

**Comment by cy6erninja** [comment]
According to [this documentation page](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable), the correct redirect should be similar to this example(worked for me). 
```yaml
steps:
  - name: Set the value
    id: step_one
    run: |
      echo "action_state=yellow" >> $GITHUB_ENV
  - name: Use the value
    id: step_two
    run: |
      echo "${{ env.action_state }}" # This will output 'yellow'
   ```
 The Environment Files [announcement page](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/) is very misleading.

**Comment by michael-s-grant** [comment]
I am still seeing this problem (in one and one only of in excess of thirty places where I have been replacing `set-output`). The above comments suggested I needed to be using a runner of at least v. 2.297.0, but mine is already 2.299.1. Placing quotes around `$GITHUB_OUTPUT` changed the error to " : no such file or directory"; the space before the semicolon indicates the environment variable was being interpreted as empty, even though debug proved it wasn't.

In the end, I worked around the problem by assigning the environment variable to second variable, and then appending using this and `tee -a`.

I have placed a comment at [this issue](https://github.com/nektos/act/issues/1442) in the hope this leads to the problem gets solved properly.
